src/statistician.jl

```diff
@@ -0,0 +1,101 @@
+"""
+    Statistician()
+
+Neural Statistician model.
+Acts on bag data (sets of instances).
+
+There are six parts:
+- instance encoder
+- context prior (isotropic Gaussian)
+- context encoder q(c|D)
+- conditional p(z|c)
+- latent instance encoder q(z|x,c)
+- decoder p(x|z)
+"""
+
+struct Statistician{IE,pc <: ContinuousMultivariateDistribution,qc <: ConditionalMvNormal,pz <: ConditionalMvNormal,qz <: ConditionalMvNormal,D <: ConditionalMvNormal} # <: AbstractNS
+	instance_encoder::IE
+	prior_c::pc
+	encoder_c::qc
+	conditional_z::pz
+	encoder_z::qz
+	decoder::D
+end
+
+Flux.@functor Statistician
+
+function Flux.trainable(m::Statistician)
+	(instance_encoder = m.instance_encoder, encoder_c = m.encoder_c, conditional_z = m.conditional_z, encoder_z = m.encoder_z, decoder = m.decoder)
+end
+
+function Statistician(clength::Int, IE, enc_c::ConditionalMvNormal, cond_z::ConditionalMvNormal, enc_z::ConditionalMvNormal, dec::ConditionalMvNormal)
+	W = first(Flux.params(enc_c))
+    μ_c = fill!(similar(W, clength), 0)
+    σ_c = fill!(similar(W, clength), 1)
+    prior_c = DistributionsAD.TuringMvNormal(μ_c, σ_c)
+	Statistician(IE, prior_c, enc_c, cond_z, enc_z, dec)
+end
+
+
+"""
+    lossNS(x::AbstractArray,model::Statistician; β1=1.0, β2=1.0)
+
+Neural Statistician loss.
+
+For NS a Statistician model, simply create a loss
+function as
+    
+    `loss(x) = lossNS(x, NS)`
+
+The β terms scale the KLDs:
+- β1: KL[q(c|D) || p(c)]
+- β2: KL[q(z|c,x) || p(z|c)]
+"""
+
+function lossNS(x, model::Statistician;β1=1.0,β2=1.0)
```

Comment:
It would be nice to have (a short version) of the loss in the docs. I have not read the paper but it is probably the ELBO that you are optimizing here? In that case it would be nice to just rename the function to `elbo`.

All other models have their loss defined as `elbo(m::ModelName, x::AbstractArray)`, so please switch the arugments here as well.

Reply:
It is ELBO, I changed it. Still, it has to be long, because the ELBO evaluation has more parts then simple VAE.